worker/src/metrics.rs

```diff
@@ -0,0 +1,43 @@
+use sysinfo::System;
+
+pub enum MemoryMeasure {
+    Megabyte,
+    Gigabyte,
+}
+
+#[derive(Debug)]
+pub struct Metric {
+    free_memory: f64,
+    cpu_usage: f64,
+}
+
+impl Metric {
+    pub fn new(measure: MemoryMeasure) -> Metric {
+        Metric {
+            free_memory: get_memory(measure), 
+            cpu_usage: 10.0
+         }
+    }
+}
+
+fn get_memory(measure: MemoryMeasure) -> f64 {
+    match measure {
+        MemoryMeasure::Gigabyte => get_memory_as_gigabyte(),
+        MemoryMeasure::Megabyte => get_memory_as_megabyte(),

```

Comment:
Hmm, you could unify these functions here by creating an associated method on the `SpaceUnit` (aka `MemoryMeasure`) enum that returns the multiplicative factor that converts bytes into the desired unit.

For example, `MiB` could return `1 / (1024 ** 2)` and `GiB` could return `1 / (1024 ** 3)`. And then, `(system.total_memory() - system.used_memory()) * unit.byte_conv_factor()`.